JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeRoute.kt

```diff
@@ -210,3 +204,40 @@ private enum class HomeScreenType {
     Feed,
     ArticleDetails
 }
+
+@Composable
+private fun getHomeScreenType(
+    windowSize: WindowSize,
+    uiState: HomeUiState
+): HomeScreenType = when (windowSize) {
+    WindowSize.Compact,
+    WindowSize.Medium -> {
+        when (uiState) {
+            is HomeUiState.HasPosts -> {
+                if (uiState.isArticleOpen) {
+                    HomeScreenType.ArticleDetails
+                } else {
+                    HomeScreenType.Feed
+                }
+            }
+            is HomeUiState.NoPosts -> HomeScreenType.Feed
+        }
+    }
+    WindowSize.Expanded -> HomeScreenType.FeedWithArticleDetails
+}
+
+@SuppressLint("ComposableModifierFactory", "ModifierFactoryExtensionFunction")
+@Composable
+private fun getScreenModifier(showNavRail: Boolean): Modifier = if (showNavRail) {
+    Modifier
+        .statusBarsPadding()
+        .padding(
+            rememberInsetsPaddingValues(
+                insets = LocalWindowInsets.current.navigationBars,
+                applyStart = false,
+                applyBottom = false
+            )
+        )
+} else {
+    Modifier
```

Comment:
When we are not showing the navrail, we still need to apply the insets to the home screens:

We want to apply the bottom system bars to both the list and the detail.
When we are only showing one, we want to apply both the start and end system bars
When we are showing both, we want to apply the start to the list (but not the end), and we want to apply the end to the detail (but not the start).

Reply:
I prefer to cover these insets work on #655 since it changes a lot things